### PR TITLE
ocp4: Use ScanSettingBindings for e2e tests

### PR DESCRIPTION
**Pull request by JAORMX**
These are the objects we recommend folks to use instead of raw
ComplianceSuites. It also allows us for auto-detection of platform/node
checks.

### CODE REVIEW

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now dd56b9d5b7bb2033d8aca4fa6e4c57a8fb03839d

File: tests/ocp4e2e/helpers.go [hunk 1]
```diff
@@ -292,92 +249,135 @@ func (ctx *e2econtext) waitForOperatorToBeReady() {
...
+		},
 	}
 
-	// We shouldn't get here.
-	return nil
+	bo := backoff.WithMaxRetries(backoff.NewConstantBackOff(apiPollInterval), 180)
+	err := backoff.RetryNotify(func() error {
+		return ctx.dynclient.Create(goctx.TODO(), binding)
+	}, bo, func(err error, d time.Duration) {
+		fmt.Printf("Couldn't create binding after %s: %s\n", d.String(), err)
```
Comment (jhrozek) [inline tests/ocp4e2e/helpers.go:366]: Is %s OK to use for formatting an error? I've usually seen %v being used.

Comment (jhrozek) [reply tests/ocp4e2e/helpers.go:366]: Hmm, git grep across some code shows it is, please disregard.

File: tests/ocp4e2e/helpers.go [hunk 2]
```diff
@@ -443,9 +443,68 @@ func (ctx *e2econtext) waitForMachinePoolUpdate(name string) error {
...
+	var lastErr error
+	err = wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+		suite := &cmpv1alpha1.ComplianceSuite{}
+		key := types.NamespacedName{Name: s, Namespace: ctx.OperatorNamespacedName.Namespace}
+		lastErr = ctx.dynclient.Get(goctx.TODO(), key, suite)
+		if lastErr != nil {
+			return false, nil
+		}
+		if suite.Status.Phase != cmpv1alpha1.PhasePending &&
+			suite.Status.Phase != cmpv1alpha1.PhaseLaunching {
```
Comment (jhrozek) [inline tests/ocp4e2e/helpers.go:487]: Hmm this sounds racy, what if the phase goes to running very fast? Wouldn't it be more robust to just check that the state changes after applying the annotation?

**Review by jhrozek — COMMENTED** [review]
Some nits inline, mostly looks good

File: tests/ocp4e2e/helpers.go [hunk 3]
```diff
@@ -292,92 +249,135 @@ func (ctx *e2econtext) waitForOperatorToBeReady() {
...
+		found := &cmpv1alpha1.ProfileBundle{}
+		if err := ctx.dynclient.Get(goctx.TODO(), key, found); err != nil {
+			if errors.IsNotFound(err) {
+				return ctx.dynclient.Create(goctx.TODO(), pb)
+			}
+			return err
+		}
+		// Update the spec in case it differs
+		found.Spec = pb.Spec
+		return ctx.dynclient.Update(goctx.TODO(), found)
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply tests/ocp4e2e/helpers.go:279]: It is, we don't have a cached client enabled here

File: tests/ocp4e2e/helpers.go (hunk 2 above)
Comment (JAORMX) [reply tests/ocp4e2e/helpers.go:487]: Should I just check that the phase is not "done" then?

Comment (jhrozek) [reply tests/ocp4e2e/helpers.go:487]: I think that would be best. Pending can be extremely fast, Launching should have some delay but...
Checking for != Done seems like safest.

File: tests/ocp4e2e/helpers.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jhrozek) [reply tests/ocp4e2e/helpers.go:279]: Ah, sure, dynclient.

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 5a26d848ddcfed55b27198d64a9893a4e6395964

**Review by jhrozek — APPROVED** [review]
/lgtm

**Pull request merged**